api/src/task/notifications/constants.py

```diff
@@ -0,0 +1,25 @@
+from dataclasses import dataclass
+from enum import StrEnum
+from typing import List
+
+
+class NotificationReasons(StrEnum):
```

Comment:
Nit: Singular as something has a single notification reason
```suggestion
class NotificationReason(StrEnum):
```